xc/device.ml

```diff
@@ -283,6 +285,37 @@ end
 (****************************************************************************************)
 (** Disks:                                                                              *)
 
+module QMP = struct
+	module Events = struct
+		type event = { domid: int; event: Qmp.event }
+		let qmp_events = Queue.create ()
+		let m, c = Mutex.create(), Condition.create ()
+		let add domid event = Mutex.execute m (fun ()-> Queue.add {domid; event} qmp_events; Condition.signal c)
+		let next () =
+			Mutex.execute m ( fun ()->
+				while Queue.is_empty qmp_events do
+					Condition.wait c m
```

Comment:
Igor had a look at the qemu code and found out that we can have an independent connection that has negotiated to receive only events. This means that we will never receive a Success/Error message from another connection, simplifying the implementation of this event loop.